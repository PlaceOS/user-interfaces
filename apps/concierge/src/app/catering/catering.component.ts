import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { BaseClass } from '@placeos/common';

@Component({
    selector: '[app-catering]',
    template: `
        <sidebar></sidebar>
        <main class="relative w-full flex flex-col">
            <catering-topbar class="relative z-10"></catering-topbar>
            <ng-container [ngSwitch]="page">
                <catering-order-list
                    *ngSwitchCase="'orders'"
                    class="flex-1"
                ></catering-order-list>
                <catering-menu
                    *ngSwitchCase="'menu'"
                    class="flex-1"
                ></catering-menu>
                <div
                    *ngSwitchDefault
                    class="flex flex-1 items-center justify-center"
                >
                    <a
                        matRipple
                        class="rounded flex flex-col items-center bg-white shadow text-black"
                        [routerLink]="['/catering', 'menu']"
                    >
                        <div
                            name="img"
                            class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                            [style.background-image]="'url(assets/menus.jpg)'"
                        >
                            <div
                                class="absolute inset-0 bg-black opacity-60 z-0"
                            ></div>
                            <span class="z-10">Menus and Pricing</span>
                        </div>
                        <div
                            class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                        >
                            View and Edit Menus and Pricing
                        </div>
                    </a>
                    <a
                        matRipple
                        class="rounded flex flex-col items-center bg-white shadow text-black"
                        [routerLink]="['/catering', 'orders']"
                    >
                        <div
                            name="img"
                            class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                            [style.background-image]="'url(assets/orders.jpg)'"
                        >
                            <div
                                class="absolute inset-0 bg-black opacity-60 z-0"
                            ></div>
                            <span class="z-10">Today's Orders</span>
                        </div>
                        <div
                            class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                        >
                            View Catering Orders and their statuses upon arrival
                        </div>
                    </a>
                </div>
            </ng-container>
        </main>
    `,
    styles: [
        `
            :host {
                display: flex;
                height: 100%;
                width: 100%;
                background: var(--bg);
            }

            a {
                width: 28rem;
                height: 16rem;
                margin-left: 0.5rem;
                text-decoration: none;
                transition: background 200ms;
            }

            a:hover {
                background-color: #00000001 !important;
            }

            a:first-child {
                margin: 0;
            }
        `,
    ],
})
export class CateringComponent extends BaseClass implements OnInit {
    /** Page being displayed */
    public page: string;

    constructor(private _route: ActivatedRoute) {
        super();
    }

    public ngOnInit() {
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }
}
