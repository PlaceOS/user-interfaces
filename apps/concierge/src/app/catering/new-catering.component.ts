import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { AsyncHandler } from '@placeos/common';

@Component({
    selector: '[app-new-catering]',
    template: `
        <app-topbar></app-topbar>
        <div class="flex flex-1 h-px">
            <app-sidebar></app-sidebar>
            <main class="flex flex-col flex-1 w-1/2 h-full">
                <catering-topbar class="relative z-10"></catering-topbar>
                <div class="flex flex-col px-8 flex-1 h-1/2">
                    <div
                        class="p-2 flex items-center justify-center bg-info text-white text-sm mb-4 rounded"
                        *ngIf="page === 'menu'"
                    >
                        To edit the menu de-select all levels from the top left
                    </div>
                    <div class="flex flex-1 h-1/2 w-full overflow-auto">
                        <ng-container [ngSwitch]="page">
                            <catering-order-list
                                *ngSwitchCase="'orders'"
                                class="flex-1"
                            ></catering-order-list>
                            <catering-menu
                                *ngSwitchCase="'menu'"
                                class="flex-1"
                            ></catering-menu>
                            <div
                                *ngSwitchDefault
                                class="flex flex-1 flex-wrap items-center justify-center"
                            >
                                <a
                                    matRipple
                                    class="rounded flex flex-col items-center bg-base-100 shadow text-black"
                                    [routerLink]="['/catering', 'menu']"
                                >
                                    <div
                                        name="img"
                                        class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                                        [style.background-image]="
                                            'url(assets/menus.jpg)'
                                        "
                                    >
                                        <div
                                            class="absolute inset-0 bg-neutral opacity-60 z-0"
                                        ></div>
                                        <span class="z-10"
                                            >Menus and Pricing</span
                                        >
                                    </div>
                                    <div
                                        class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                                    >
                                        View and Edit Menus and Pricing
                                    </div>
                                </a>
                                <a
                                    matRipple
                                    class="rounded flex flex-col items-center bg-base-100 shadow text-black"
                                    [routerLink]="['/catering', 'orders']"
                                >
                                    <div
                                        name="img"
                                        class="relative w-full flex flex-1 items-center justify-center text-2xl text-white bg-center bg-cover"
                                        [style.background-image]="
                                            'url(assets/orders.jpg)'
                                        "
                                    >
                                        <div
                                            class="absolute inset-0 bg-neutral opacity-60 z-0"
                                        ></div>
                                        <span class="z-10">Today's Orders</span>
                                    </div>
                                    <div
                                        class="p-2 h-14 w-full text-sm text-center flex items-center justify-center"
                                    >
                                        View Catering Orders and their statuses
                                        upon arrival
                                    </div>
                                </a>
                            </div>
                        </ng-container>
                    </div>
                </div>
            </main>
        </div>
    `,
    styles: [
        `
            :host {
                display: flex;
                flex-direction: column;
                height: 100%;
                width: 100%;
                background-color: var(--b1);
            }

            a {
                width: 28rem;
                height: 16rem;
                margin-left: 0.5rem;
                text-decoration: none;
                transition: background 200ms;
            }

            a:hover {
                opacity: 0.8;
            }

            a:first-child {
                margin: 0;
            }
        `,
    ],
})
export class NewCateringComponent extends AsyncHandler implements OnInit {
    /** Page being displayed */
    public page: string;

    constructor(private _route: ActivatedRoute) {
        super();
    }

    public ngOnInit() {
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }
}
