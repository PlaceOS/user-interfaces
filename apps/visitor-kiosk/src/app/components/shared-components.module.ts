import { NgModule, Type } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { ScrollingModule } from '@angular/cdk/scrolling';
import { EditorModule } from '@tinymce/tinymce-angular';

import { MatButtonModule } from '@angular/material/button';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatDialogModule } from '@angular/material/dialog';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { MatAutocompleteModule } from '@angular/material/autocomplete';
import { MatSelectModule } from '@angular/material/select';
import { MatChipsModule } from '@angular/material/chips';
import { MatRippleModule } from '@angular/material/core';
import { MatMenuModule } from '@angular/material/menu';
import { MatBottomSheetModule } from '@angular/material/bottom-sheet';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatRadioModule } from '@angular/material/radio';

import { ComponentsModule } from '@placeos/components';

import { TopbarHeaderComponent } from './topbar-header.component';
import { TakePhotoComponent } from './take-photo.component';

const COMPONENTS: Type<any>[] = [TopbarHeaderComponent, TakePhotoComponent];

const MATERIAL_MODULES: any[] = [
    MatButtonModule,
    MatSnackBarModule,
    MatDialogModule,
    MatFormFieldModule,
    MatInputModule,
    MatProgressSpinnerModule,
    MatProgressBarModule,
    MatAutocompleteModule,
    MatSelectModule,
    MatChipsModule,
    MatRippleModule,
    MatMenuModule,
    MatBottomSheetModule,
    MatCheckboxModule,
    MatRadioModule,
];

const CDK_MODULES: any[] = [ScrollingModule];

@NgModule({
    declarations: [...COMPONENTS],
    imports: [
        CommonModule,
        FormsModule,
        ReactiveFormsModule,
        ComponentsModule,
        RouterModule,
        EditorModule,
        ...MATERIAL_MODULES,
        ...CDK_MODULES,
    ],
    exports: [
        ComponentsModule,
        ...COMPONENTS,
        ...MATERIAL_MODULES,
        ...CDK_MODULES,
    ],
})
export class SharedComponentsModule {}
