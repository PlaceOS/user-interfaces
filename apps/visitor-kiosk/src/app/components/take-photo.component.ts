import {
    Component,
    OnInit,
    HostBinding,
    OnDestroy,
    Output,
    EventEmitter,
} from '@angular/core';

@Component({
    selector: 'a-take-photo',
    template: `
        <div
            name="camera"
            class="flex flex-col items-center justify-center rounded overflow-hidden relative"
        >
            <mat-spinner diameter="32"></mat-spinner>
            <div class="text">Please wait...</div>
            <video
                id="video"
                #video
                autoplay
                width="360"
                height="400"
                class="absolute inset-0"
            ></video>
            <canvas
                id="canvas"
                #canvas
                class="absolute inset-0"
                width="360"
                height="400"
            ></canvas>
        </div>
        <div class="flex items-center justify-center mt-4 space-x-2">
            <button
                class="take-photo"
                *ngIf="!hasPhoto; else accept_state"
                mat-button
                (click)="takePhoto()"
            >
                Take Photo
            </button>
        </div>
        <ng-template #accept_state>
            <button class="inverse" mat-button (click)="cancelPhoto()">
                Cancel
            </button>
            <button mat-button (click)="acceptPhoto()">Accept</button>
        </ng-template>
    `,
    styles: [
        `
            :host {
                display: flex;
                flex-direction: column;
                align-items: center;
            }

            [name='camera'] {
                width: 22.5rem;
                height: 25rem;
            }

            button {
                width: 8rem;
            }
        `,
    ],
})
export class TakePhotoComponent implements OnInit, OnDestroy {
    @Output() public photoAccepted = new EventEmitter();
    @HostBinding('class.has-photo') public hasPhoto = false;
    public video: HTMLVideoElement;
    public canvas: HTMLCanvasElement;
    private canvasContext = null;

    private constraints = {
        audio: false,
        video: {
            width: { min: 360, max: 360 },
            height: { min: 400, max: 400 },
        },
    };

    public imgSrc = null;

    public ngOnInit() {
        setTimeout(() => this.getCanvasContext(), 500);
    }

    public ngOnDestroy() {
        this.stopCapture();
        this.video = null;
        this.canvasContext = null;
        this.canvas = null;
        this.imgSrc = null;
    }

    private startCapture() {
        this.imgSrc = null;
        console.log('TakePhotoComponent::startCapture');
        this.video = document.getElementById('video') as any;
        if (!this.video) {
            return setTimeout(() => this.startCapture(), 200);
        } else {
            navigator.mediaDevices
                .getUserMedia(this.constraints)
                .then((stream) => {
                    this.video.srcObject = stream;
                });
        }
    }

    private stopCapture() {
        console.log('TakePhotoComponent::stopCapture');
        if (this.video?.srcObject) {
            (this.video.srcObject as any)
                .getVideoTracks()
                .forEach((track) => track.stop());
        }
    }

    private getCanvasContext() {
        console.log('TakePhotoComponent::getCanvasContext');
        this.canvas = document.getElementById('canvas') as any;
        if (!this.canvas) {
            return setTimeout(() => this.getCanvasContext(), 200);
        }
        this.canvasContext = this.canvas.getContext('2d');
        setTimeout(() => this.startCapture(), 200);
    }

    public takePhoto() {
        console.log('TakePhotoComponent::takePhoto');
        this.canvasContext.drawImage(
            this.video,
            0,
            0,
            this.canvas.width,
            this.canvas.height
        );
        this.hasPhoto = true;
        this.stopCapture();
    }

    public cancelPhoto() {
        console.log('TakePhotoComponent::cancelPhoto');
        this.canvasContext.clearRect(
            0,
            0,
            this.canvas.width,
            this.canvas.height
        );
        this.hasPhoto = false;
        this.startCapture();
    }

    public acceptPhoto() {
        try {
            this.imgSrc = this.canvas.toDataURL('image/jpeg', 0.75);
            this.photoAccepted.emit(this.imgSrc);
            console.log('TakePhotoComponent::acceptPhoto Success');
        } catch (err) {
            console.error(
                'TakePhotoComponent::acceptPhoto Error converting image',
                err
            );
            this.photoAccepted.emit(null);
            this.cancelPhoto();
        }
    }
}
