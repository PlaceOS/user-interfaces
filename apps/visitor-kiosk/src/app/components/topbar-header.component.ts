import { Component } from '@angular/core';
import { format } from 'date-fns';

@Component({
    selector: 'a-topbar-header',
    template: `
        <div class="flex items-center justify-between bg-secondary shadow">
            <a
                mat-button
                [routerLink]="['/']"
                class="h-full flex flex-col justify-center px-4"
            >
                <img class="h-10 my-2" src="assets/logo-dark.svg" />
            </a>
            <div
                class="ml-auto h-full flex flex-col justify-center text-white px-4"
            >
                {{ time }}
            </div>
        </div>
    `,
    styles: [
        `
            :host > div {
                height: 3.5rem;
            }

            a {
                border-radius: 0;
            }

            img {
                max-height: 2.5rem;
                max-width: 50vw;
            }
        `,
    ],
})
export class TopbarHeaderComponent {
    public date: number;

    public get time() {
        return format(this.date || Date.now(), 'EEEE, MMMM do, h:mma');
    }
}
