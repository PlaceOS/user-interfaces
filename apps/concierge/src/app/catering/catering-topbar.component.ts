import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { first, map, take } from 'rxjs/operators';

import { AsyncHandler, SettingsService } from '@placeos/common';
import { OrganisationService } from '@placeos/organisation';
import {
    CateringOrdersService,
    CateringStateService,
    ChargeCodeListModalComponent,
} from '@placeos/catering';
import { MatDialog } from '@angular/material/dialog';
import { AvailableRoomsStateModalComponent } from '@placeos/components';
import { combineLatest } from 'rxjs';

@Component({
    selector: 'catering-topbar',
    template: `
        <div class="flex items-center w-full pt-4 pb-2 px-8 space-x-2">
            <h2 class="text-2xl font-medium">
                Catering {{ page === 'menu' ? 'Menu' : 'Orders' }}
            </h2>
            <div class="flex-1 w-px"></div>
            <searchbar
                class="mr-2"
                [model]="filters?.search"
                (modelChange)="setSearch($event)"
            ></searchbar>
        </div>
        <div class="flex items-center bg-base-100 h-20 px-8 space-x-2">
            <mat-form-field appearance="outline" class="no-subscript w-60">
                <mat-select
                    [ngModel]="filters?.zones"
                    (ngModelChange)="updateZones($event)"
                    placeholder="All Levels"
                    multiple
                >
                    <mat-option
                        *ngFor="let level of levels | async"
                        [value]="level.id"
                    >
                        <div class="flex flex-col-reverse">
                            <div class="text-xs opacity-30" *ngIf="use_region">
                                {{ (level.parent_id | building)?.display_name }}
                                <span class="opacity-0"> - </span>
                            </div>
                            <div>
                                {{ level.display_name || level.name }}
                            </div>
                        </div>
                    </mat-option>
                </mat-select>
            </mat-form-field>
            <div *ngIf="page === 'menu'" class="flex-1 w-2"></div>
            <button
                *ngIf="
                    page === 'menu' && (!zones[0] || zones[0] === building?.id)
                "
                icon
                matRipple
                matTooltip="Add Item"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="addItem()"
            >
                <app-icon class="text-2xl">add</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Edit Config"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="editConfig()"
            >
                <app-icon class="text-2xl">menu_book</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Import Menu"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="importMenu()"
            >
                <app-icon class="text-2xl">cloud_upload</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Room Availability"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="setRoomAvailability()"
            >
                <app-icon class="text-2xl">event_available</app-icon>
            </button>
            <button
                *ngIf="page === 'menu'"
                icon
                matRipple
                matTooltip="Charge Codes"
                class="bg-secondary text-secondary-content rounded h-12 w-12"
                (click)="setChargeCodes()"
            >
                <app-icon class="text-2xl">payments</app-icon>
            </button>
            <div *ngIf="page !== 'menu'" class="flex-1 w-2"></div>
            <!-- <searchbar class="mr-2"></searchbar> -->
            <date-options
                *ngIf="page !== 'menu'"
                (dateChange)="setDate($event)"
            ></date-options>
        </div>
    `,
    styles: [
        `
            mat-form-field {
                height: 3.25em;
                width: 8em;
            }
        `,
    ],
})
export class CateringTopbarComponent extends AsyncHandler implements OnInit {
    /** List of selected levels */
    public zones: string[] = [];
    /** Currently active page */
    public page: string;
    public readonly filters = this._orders.filters;
    /** List of levels for the active building */
    public readonly levels = combineLatest([
        this._org.active_building,
        this._org.active_region,
    ]).pipe(
        map(([bld, region]) =>
            this._settings.get('app.use_region')
                ? this._org.levelsForRegion(region)
                : this._org.levelsForBuilding(bld)
        )
    );
    /** Set filtered date */
    public readonly setDate = (date) =>
        (this._orders.filters = { ...this._orders.filters, date });
    public readonly setSearch = (str) =>
        (this._orders.filters = { ...this._orders.filters, search: str });
    /** List of levels for the active building */
    public readonly updateZones = (z) => {
        this._router.navigate([], {
            relativeTo: this._route,
            queryParams: { zone_ids: z.join(',') },
        });
        this._orders.filters = { ...this._orders.filters, zones: [z] };
        this._catering.zone = z[0];
    };

    public readonly addItem = () => this._catering.addItem();
    public readonly editConfig = () => this._catering.editConfig();
    public readonly importMenu = () => this._catering.importMenu();

    public get building() {
        return this._org.building;
    }

    public get use_region() {
        return !!this._settings.get('app.use_region');
    }

    constructor(
        private _orders: CateringOrdersService,
        private _catering: CateringStateService,
        private _org: OrganisationService,
        private _route: ActivatedRoute,
        private _router: Router,
        private _dialog: MatDialog,
        private _settings: SettingsService
    ) {
        super();
    }

    public async ngOnInit() {
        await this._org.initialised.pipe(first((_) => _)).toPromise();
        this._catering.zone =
            (this._orders.filters?.zones || [])[0] || this._org.building?.id;
        this.subscription(
            'route.query',
            this._route.queryParamMap.subscribe((params) => {
                if (params.has('zone_ids')) {
                    const zones = params.get('zone_ids').split(',');
                    if (!zones.length) return;
                    const level = this._org.levelWithID(zones);
                    this.zones = zones;
                    if (!level) return;
                    this._org.building = this._org.buildings.find(
                        (bld) => bld.id === level.parent_id
                    );
                }
            })
        );
        this.subscription(
            'route.params',
            this._route.paramMap.subscribe(
                (params) =>
                    (this.page = params.has('view') ? params.get('view') : '')
            )
        );
    }

    public async setRoomAvailability() {
        const ref = this._dialog.open(AvailableRoomsStateModalComponent, {
            data: {
                type: 'Catering',
                disabled_rooms: await this._catering.availability
                    .pipe(take(1))
                    .toPromise(),
            },
        });
        this.subscription(
            'room-availability',
            ref.componentInstance.change.subscribe(async (list) => {
                await this._catering
                    .saveSettings({ disabled_rooms: list })
                    .catch();
                ref.componentInstance.loading = false;
            })
        );
    }

    public setChargeCodes() {
        this._dialog.open(ChargeCodeListModalComponent);
    }
}
