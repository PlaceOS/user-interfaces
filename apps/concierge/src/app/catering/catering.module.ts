import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { RouterModule, Route } from '@angular/router';

import { MatTabsModule } from '@angular/material/tabs';
import { MatChipsModule } from '@angular/material/chips';

import { UIModule } from '../ui/ui.module';

import { CateringComponent } from './catering.component';
import { CateringTopbarComponent } from './catering-topbar.component';
import { SharedCateringModule } from '@placeos/catering';

const ROUTES: Route[] = [
    { path: '', component: CateringComponent },
    { path: ':view', component: CateringComponent },
];

@NgModule({
    declarations: [CateringComponent, CateringTopbarComponent],
    imports: [
        CommonModule,
        FormsModule,
        UIModule,
        MatTabsModule,
        MatChipsModule,
        SharedCateringModule,
        RouterModule.forChild(ROUTES),
    ],
})
export class CateringModule {}
