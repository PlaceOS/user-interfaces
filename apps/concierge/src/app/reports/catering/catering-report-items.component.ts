import { Component } from '@angular/core';
import { CateringOption } from '@placeos/catering';
import { CateringReportStateService } from './catering-report-state.service';

@Component({
    selector: 'catering-report-items',
    template: `
        <div
            class="w-[64rem] max-w-[calc(100%-2rem)] rounded overflow-hidden bg-white shadow mx-auto my-2"
        >
            <div
                class="border-b border-gray-200 flex items-center justify-between px-4"
            >
                <h2 class="py-2 text-xl font-medium">Ordered Items</h2>
            </div>
            <custom-table
                class="w-full h-full"
                [dataSource]="items"
                [columns]="[
                    'name',
                    'options',
                    'quantity',
                    'unit_price',
                    'total_cost'
                ]"
                [display_column]="[
                    'Name',
                    'Options',
                    'Quantity',
                    'Unit Price',
                    'Total Cost'
                ]"
                [column_size]="['flex']"
                [template]="{
                    options: option_state,
                    unit_price: cost_state,
                    total_cost: cost_state
                }"
                [pagination]="true"
                empty="No orders for selected period"
            ></custom-table>
            <ng-template #option_state let-data="data">
                <span
                    class="text-xs px-2 py-1 rounded bg-gray-300"
                    *ngIf="data.length"
                    [matTooltip]="options(data)"
                >
                    {{ data.length }} option(s)
                </span>
            </ng-template>
            <ng-template #cost_state let-data="data">
                {{ (data || 0) / 100 | currency }}
            </ng-template>
        </div>
    `,
    styles: [``],
})
export class CateringReportItemsComponent {
    public readonly items = this._report.catering_items;

    constructor(private _report: CateringReportStateService) {}

    public options(opts: CateringOption[]) {
        return opts.map((_) => _.name).join('\n');
    }
}
