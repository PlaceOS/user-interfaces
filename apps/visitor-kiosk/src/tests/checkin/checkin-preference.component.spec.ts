import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatSelectModule } from '@angular/material/select';
import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import { CateringStateService } from '@placeos/catering';
import { IconComponent } from '@placeos/components';
import { MockComponent } from 'ng-mocks';
import { of } from 'rxjs';

import { CheckinPreferencesComponent } from '../../app/checkin/checkin-preferences.component';
import { CheckinStateService } from '../../app/checkin/checkin-state.service';

describe('CheckinPreferencesComponent', () => {
    let spectator: SpectatorRouting<CheckinPreferencesComponent>;
    const createComponent = createRoutingFactory({
        component: CheckinPreferencesComponent,
        declarations: [MockComponent(IconComponent)],
        providers: [
            {
                provide: CheckinStateService,
                useValue: { event: of({}), guest: of({}) },
            },
            {
                provide: CateringStateService,
                useValue: { menu: of([]) },
            },
        ],
        imports: [MatFormFieldModule, MatSelectModule, FormsModule, ReactiveFormsModule],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });
});
