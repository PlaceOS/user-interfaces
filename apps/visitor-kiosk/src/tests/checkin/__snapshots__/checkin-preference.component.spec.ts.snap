// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckinPreferencesComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="bg-white rounded shadow overflow-hidden relative flex flex-col items-center my-4 p-4 mx-auto"
  >
    <h3
      class="text-xl mb-2 w-full"
    >
      Would you like a drink?
    </h3>
    <div
      class="w-full"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field w-full mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float ng-untouched ng-pristine ng-valid _mat-animation-noopable"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="mat-select mat-select-empty ng-untouched ng-pristine ng-valid"
                placeholder="Select beverage"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    <span
                      class="mat-select-placeholder mat-select-min-line"
                    >
                      Select beverage
                    </span>
                    
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
            
          </div>
        </div>
      </mat-form-field>
    </div>
    <button
      class="w-32"
      mat-button=""
    >
       Continue 
    </button>
    <a
      class="absolute top-0 right-0"
      mat-icon-button=""
    >
      <app-icon>
        close
      </app-icon>
    </a>
  </div>
</div>
`;
